### ISSUE TITLE
Ltsp Mint xfce 19.2 IPXE could not boot image

### ISSUE THREAD

**Issue by doguibnu** [ltsp/ltsp#21]

**Comment by doguibnu** [comment]
Hello!

I start new test with mint xfce 19.2 64 bits and after install LTSP (following the steps), when client boot, IPXE show the message: 
Link up
TXE: 1 x  "Network unreachable
Configuring .....ok
net0: 192.168.67:69/255.255.255.0 gw 192.168.67.1
next server: 192.168.67.1
filename: ltsp/ltsp.ipxe
tftp://192.168.67.1/ltsp/ltsp.ipxe....ok
ltsp.ipxe: 2987 bytes [script]
could not boot image: Result too large 
No more network devices 

and stop boot

what can be wrong? 

Thanks so much

Douglas

**Comment by alkisg** [comment]
Can you paste the output of the following commands?

```shell
find /srv/tftp/ -ls
cat /srv/tftp/ltsp/ltsp.ipxe
```

**Comment by doguibnu** [comment]
> Can you paste the output of the following commands?
> 
> ```shell
> find /srv/tftp/ -ls
> cat /srv/tftp/ltsp/ltsp.ipxe
> ```

root@ltspserver:~# find /srv/tftp/ -ls
  4596290      4 drwxr-xr-x   3 root     root         4096 set  2 11:00 /srv/tftp/
  4867294      4 drwxr-xr-x   3 root     root         4096 set  2 11:03 /srv/tftp/ltsp
  4867295      4 -rw-r--r--   1 root     root         2987 set  2 11:01 /srv/tftp/ltsp/ltsp.ipxe
  1704090    136 -rw-r--r--   1 root     root       138240 set  2 11:03 /srv/tftp/ltsp/ltsp.img
  4867297   1208 -rw-r--r--   1 root     root      1235864 jul 24 07:10 /srv/tftp/ltsp/memtest.efi
  4867296    160 -rw-r--r--   1 root     root       161968 jul 24 07:10 /srv/tftp/ltsp/memtest.0
  4867298    156 -rw-r--r--   1 root     root       159296 jul 24 07:10 /srv/tftp/ltsp/snponly.efi
  4992054      4 drwxr-xr-x   2 root     root         4096 set  2 11:33 /srv/tftp/ltsp/x86_64
  4992056  59188 -rw-r--r--   1 root     root     60607132 set  2 10:57 /srv/tftp/ltsp/x86_64/initrd.img
  4992055   8136 -rw-r--r--   1 root     root      8330904 ago 22 13:37 /srv/tftp/ltsp/x86_64/vmlinuz
  4867299     68 -rw-r--r--   1 root     root        67642 jul 24 07:10 /srv/tftp/ltsp/undionly.kpxe
root@ltspserver:~# 


cat /srv/tftp/ltsp/ltsp.ipxe
#!ipxe
# This file is part of LTSP, https://ltsp.github.io
# Copyright 2019 the LTSP team, see AUTHORS
# SPDX-License-Identifier: GPL-3.0-or-later

# Configure iPXE for LTSP
# Documentation=man:ltsp-ipxe(8)

# Set the default image (img) based on arch, or to root-path if it's not empty
cpuid --ext 29 && set img x86_64 || set img x86_32
isset ${root-path} && set img ${root-path} ||
isset ${proxydhcp/dhcp-server} && set srv ${proxydhcp/dhcp-server} || set srv ${next-server}

# Define cmdline parameters common to all LTSP clients
set cmdline_ltsp BOOTIF=01-${mac:hexhyp}

goto ${mac} || goto start
# The following client-specific settings can be defined in ltsp.conf:
# DEFAULT_IMAGE (img): default menu item for that client
# KERNEL_PARAMETERS (cmdline_client): additional kernel parameters
# MENU_TIMEOUT (menu-timeout): menu milliseconds, 0=forever, -1=hide menu
# For those clients, :mac:address sections are generated below

:61:6c:6b:69:73:67

:start
# To completely hide the menu, set menu-timeout to -1
isset ${menu-timeout} || set menu-timeout 5000
iseq "${menu-timeout}" "-1" && goto ${img}
menu iPXE boot menu - ${hostname}:${srv}:${root-path} || goto ${img}
item --gap                        Boot an image from the network in LTSP mode:
item --key 1 x86_64               x86_64.img

item
item --gap                        Other options:
item --key m memtest              Memory test
item --key c config               Enter iPXE configuration
item --key s shell                Drop to iPXE shell
item --key d disk                 Boot from the first local disk
item
item --key x exit                 Exit iPXE and continue BIOS boot
choose --timeout ${menu-timeout} --default ${img} img || goto cancel
goto ${img}

:x86_64
:images
# The "images" method can boot anything in /srv/ltsp/images
set cmdline_method root=/dev/nfs nfsroot=${srv}:/srv/ltsp ltsp.image=images/${img}.img loop.max_part=9
goto ltsp

:roots
# The "roots" method can boot all /srv/ltsp/roots
set cmdline_method root=/dev/nfs nfsroot=${srv}:/srv/ltsp/${img}
goto ltsp

:ltsp
# :images and :roots jump here after setting cmdline_method
set cmdline ${cmdline_method} ${cmdline_ltsp} ${cmdline_client}
# In EFI mode, iPXE requires initrds to be specified in the cmdline
kernel /ltsp/${img}/vmlinuz initrd=ltsp.img initrd=initrd.img ${cmdline}
initrd /ltsp/ltsp.img
initrd /ltsp/${img}/initrd.img
boot || goto failed

:memtest
iseq ${platform} pcbios && kernel memtest.0 || kernel memtest.efi
# Boot "fails" on normal memtest exit with Esc, so show the menu again
boot ||
goto start

:config
config
goto start

:shell
echo Type 'exit' to get the back to the menu
shell
goto start

:disk
# Boot the first local HDD
sanboot --no-describe --drive 0x80 || goto failed

:exit
# Exit with error, to fall back to the next boot option
# http://forum.ipxe.org/showthread.php?tid=6775
exit 1

:cancel
echo You cancelled the menu, dropping to shell
goto shell

:failed
echo Booting failed, dropping to shell
goto shell
root@ltspserver:~#

**Comment by doguibnu** [comment]
Ohhh, I am sorry that ugly post
I will try again

**Comment by doguibnu** [comment]
[cat-srv-tftp-ltsp-ltsp-ipxe.txt](https://github.com/ltsp/ltsp/files/3566761/cat-srv-tftp-ltsp-ltsp-ipxe.txt)
[find-srv-tftp-ls.txt](https://github.com/ltsp/ltsp/files/3566762/find-srv-tftp-ls.txt)

here is better

sorry

**Comment by alkisg** [comment]
Hi Douglas,

sorry, my bad, I restructured ltsp.ipxe and I forgot to put a || operator.
Fix committed in https://github.com/ltsp/ltsp/commit/72dca090ff30bba1256c3ae10b32e303dd15d3b0, a new build will be available in a while in the PPA.

To paste shell output in github, use three backticks, like this:
\`\`\`shell
output goes here
\`\`\`

I fixed it for you in your message above, if you want press [edit] to see what I did.

Thanks!
